activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -140,9 +140,15 @@ def rollback
       end
 
       def commit
-        connection.commit_db_transaction
+        commit_result = connection.commit_db_transaction
+        # commit_db_transaction will return PG::Result upon success, otherwise PG::Error
+        @records.uniq.each {|r| ship_committed_record(r)} if commit_result.is_a?(PG::Result)
```

Comment:
I assume this doesn't fire in case of a rollback?

Reply:
That's correct. It won't ship in the case of a rollback. Tested in both cases where we are trying to persist a record outside of a Rails txn and within a rollback-able Rails txn.